src/lib/zf/log.c

```diff
@@ -135,14 +136,25 @@ void zf_log(struct zf_stack* st, const char* fmt, ...)
   va_end(va);
 }
 
+void zf_log_set_file_name(const char *file)
+{
+  strncpy(zf_log_file_name, file, sizeof(zf_log_file_name));
+  zf_log_file_name[ZF_LOG_FILE_NAME_SIZE-1] = '\0';
+
+  if( strnlen(file, ZF_LOG_FILE_NAME_SIZE) > ZF_LOG_FILE_NAME_SIZE-1 ) 
+    zf_log_stack_warn(NO_STACK, "WARNING: File name has been truncated as it is longer than %i characters\n", ZF_LOG_FILE_NAME_SIZE-1);
+}
+
 int zf_log_redirect(const char* file)
 {
-  FILE* f = fopen(file, "a");
+  zf_log_set_file_name(file);
+  FILE* f = fopen(zf_log_file_name, "a");
```

Comment:
I would check opening before setting the name.  Opening could fail for several reason (e.g. permissions) so we wouldn't want the new name set.

Reply:
`zf_log_set_file_name` now can return an error if `file` exceeds the max allowed length. It'll either fail because name is not of valid length or `fopen` is not successful. I prefer checking first whether the intended file name is a valid one, otherwise if it can be `fopen`ed and it is not a valid length, then I'd need to close/delete the empty file for nothing.